\allowdisplaybreaks{
We briefly describe the proposed computational framework. 


\subsection*{\tnet Framework}
We do not assume that the mutational  variations at the individual indices of a genomic sequence are independent (See Fig~\ref{figscheme}a). Irrespective of whether mutations are truly random~\cite{hernandez2018algorithmically}, since only certain combinations of individual mutations are viable, individual mutations across a genomic sequence replicating in the wild  appear  constrained, which is what is explicitly  modeled in our approach.

% 

Consider a set of random variables $X=\{X_i\}$, with $i \in \{1, \cdots, N\}$, each taking value from the respective sets $\Sigma_i$. Here each $X_i$ is the random variable modeling the ``outcome'' $i.e.$ the AA residue at the $i^{th}$ index of the protein sequence. A sample $x \in \prod_1^N \Sigma_i$ is an ordered $N$-tuple, which is a specific strain in this context,  consisting of a realization of each of the variables $X_i$ with the $i^{th}$ entry $x_i$ being the realization of random variable $X_i$.

We use the notation $x_{-i}$ and $x^{i,\sigma}$ to denote:
\begin{subequations}\cgather{
x_{-i} \triangleq x_1, \cdots, x_{i-1},x_{i+1},\cdots,x_N\\
x^{i,\sigma} \triangleq x_1, \cdots, x_{i-1},\sigma,x_{i+1},\cdots,x_N, \sigma \in \Sigma_i
}\end{subequations} Also, $\Dx(S)$ denotes the set of probability measures on  a set $S$, $e.g.$,  $\D$ is the set of  distributions on  $\Sigma_i$.

We note that $X$ defines a random field~\cite{vanmarcke2010random} over the index set $\{1, \cdots, N\}$. 

\begin{defn}[\tnet]
For a random field $X=\{X_i\}$ indexed by $i \in \{1, \cdots, N\}$, the \tnet is defined to be the set of predictors $\Phi=\{\qn\}$, $i.e.$, we have:
\cgather{
\qn : \prod_{j \neq i} \Sigma_j \rightarrow \D,
}  where for a sequence $x$, $\Phi_i(x_{-i}) $ estimates the distribution of $X_i$ on the set $\Sigma_i$.
\end{defn}
We use conditional inference trees as models for predictors~\cite{Hothorn06unbiasedrecursive}, although more general models are possible.





\subsection*{Biology-Aware Distance Between Sequences}
The mathematical form of our metric is not arbitrary; JS divergence is a symmetricised version of the more common KL divergence~\cite{cover} between distributions, and among  different possibilities, the \qdist  is the simplest metric such that the likelihood of a spontaneous jump (See Eq.~\eqref{fundeq} in Methods) is provably bounded above and below  by simple exponential functions of the \qdist.

\begin{defn}[\qdist: adaptive biologically meaningful dissimilarity between sequences]\label{defqdistance}
Given two sequences $x,y \in \prod_1^N\Sigma_i$, such that $x,y$ are drawn from the  populations $P,Q$  inducing the \tnet $\Phi^P,\Phi^Q$, respectively,  we define a pseudo-metric $\theta(x,y) $, as follows:
\cgather{\label{q-distance}
\theta(x,y) \triangleq \mathbf{E}_i \left (  \J^{\frac{1}{2}} \left (\qn^P(x_{-i}) , \qn^Q(y_{-i})\right ) \right )
} 
where $ \J(\cdot,\cdot)$ is the Jensen-Shannon divergence~\cite{manning1999foundations} and $\mathbf{E}_i$ indicates expectation over the indices.
\end{defn}
The square-root in the definition arises naturally from the bounds we are able to prove, and is dictated by the form of Pinsker's inequality~\cite{cover}, ensuring that   the sum of the length of successive path fragments equates the length of the path.%, making it possible to use standard  algorithms  for q-phylogeny construction.


%\subsection*{Significance Test for Population Membership}
\subsection*{Membership Degree}

For our modeling to be reliable, we need a quantitative test of how well the \tnet represents the data. Here, we formulate an explicit membership test to ascertain if individual samples may indeed be generated by the \tnet with sufficiently high probability.
%
\begin{defn}[Membership probability of a sequence]\label{defmem}
Given a population $P$ inducing the \tnet $\Phi^P$ and a sequence $x$, we can compute the membership probability of $x$:
\cgather{
\mem{x}^P \triangleq Pr(x \in P) = \prod_{j=1}^N \left ( \Phi^P_j(x_{-j}) \vert_{x_j} \right )
}
\end{defn}
$x_j$ is the $j^{th}$ entry in $x$, and is thus an element in the set $\Sigma_j$. Since we are mostly concerned with the case where $\Sigma_j$ is a finite set, $\Phi^P_j(x_{-j}) \vert_{x_j}$ is the entry in the probability mass function corresponding to the element of $\Sigma_j$ which appears at the  $j^{th}$ index in sequence $x$. 
 
We can carry out this calculation for a sequence $x$  known to be in the population $P$ as well, which allows us to define the membership degree $\M^P_x$.
\begin{defn}[Membership degree]
Let $X$ be a random field representing a population $P$, $ie.$. $X=x$ is a randomly drawn sequence from $P$. Then  the membership degree $\M^P$ is  a function of the random variable $X$: 
\cgather{
\M^P(X)  \triangleq  \prod_{j=1}^N \left ( \Phi^P_j(X_{-j}) \vert_{X_j} \right )
}Note that $\M^P$ takes values in the unit interval $[0,1]$, and the probability  $x$ is a member of the population $P$ is $\M^P(X=x)$, denoted briefly as $\mem{x}^P$ or $\mem{x}$ if $P$ is clear from context.
\end{defn}
Since $\M^P(X)$ is a random variable, we can now compute sets of sequences that better represent the population $P$, and ones that are on the fringe. We can also evaluate using a pre-specified significance-level if a particular sequence is not from the population $P$.



\subsection*{Theoretical Probability Bounds}

The \tnet framework  allows us to rigorously compute bounds on the probability of a spontaneous change of one strain to another, brought about by chance mutations. While any sequence of mutations is equally likely, the ``fitness'' of the resultant strain, or the probability that it will even result in a viable strain, or not. Thus the necessity of preserving  function  dictates that not all random changes  are viable, and the probability of observing some trajectories through the sequence space  are far greater  than others. The \tnet framework allows us to explore this constrained dynamics, as revealed by a sufficiently large set of genomic sequences.



The mathematical intuition  relating  \qdist  to the log-likelihood of spontaneous change  is similar to quantifying the  odds of  a rare biased outcome when we  toss a fair coin.
While for an unbiased coin, the odds of roughly 50\% heads is overwhelmingly likely, large deviations do happen rarely, and it turns out that the probability of such rare deviations can be explicitly quantified with existing statistical theory~\cite{varadhan2010large}.
 Generalizing to non-uniform conditional distributions inferred by the \tnet, the likelihood of a spontaneous transition  by random chance may also be similarly bounded.


We show in Theorem~\ref{thmbnd} in the supplementary text that at a significance level $\alpha$, with a sequence length $N$, the probability of spontaneous jump of sequence $x$ from population $P$ to sequence $y$ in population $Q$, $Pr(x \rightarrow y)$, is bounded by:
\cgather{\label{fundeq}
\mem{y}^Q e^{ \frac{\sqrt{8}N^2}{1-\alpha}\theta(x,y)} \geqq Pr(x \rightarrow y) \geqq \mem{y}^Q e^{-\frac{\sqrt{8}N^2}{1-\alpha}\theta(x,y)}}
where $\mem{y}^Q$ is the membership probability of strain $y$ in the target population, $N$ is the sequence length, and $\alpha$ is the statistical signifacnce level.


\subsection*{Predicting Dominant Seasonal Strains} 

Analyzing the distribution of sequences observed to circulate in the human population at the present time allows us to forecast dominant strain(s) in the next flu season as follows:

Let $\dst$ be a dominant strain in the upcoming flu season at time $t+\delta$,
where $H^t$ is the set of observed strains presently in circulation in the human population (at time $t$). We will assume that the \tnet is constructed using the sequences in teh set $H^t$, and remains unchanged upto $t+\delta$. Since this set is a function of time, the inferred \tnet also changes with time, and the induced \qdist is denoted as $\theta^{[t]}(\cdot,\cdot)$.

From the RHS bound established in Theorem~\ref{thmbnd} (See Eq.~\eqref{fundeq} above) in the supplementary text, we have:
%
\calign{
  &\ln  \frac{Pr(x \rightarrow \dsta)}{\mem{\dsta}} \geqq  -\frac{\sqrt{8}N^2}{1-\alpha}\theta^{[t]}(x,\dsta)\\
\Rightarrow &\sum_{x \in H^t} \ln  \frac{Pr(x \rightarrow \dsta)}{\mem{\dsta}}  
\geqq  \sum_{x \in H^t}-\frac{\sqrt{8}N^2}{1-\alpha}\theta^{[t]}(x,\dsta)\\
\Rightarrow  &\sum_{x\in H^t}  \theta^{[t]}(x,\dsta) - \abs{H^t}A \ln \mem{\dsta} \geqq  A \ln \frac{1}{\prod_{x \in H^t} Pr(x \rightarrow \dsta)} \intertext{where $A =\frac{1-\alpha}{\sqrt{8}N^2} $, where $N$ is the sequence length considered, and $\alpha$ is a fixed significance level. Since minimizing the LHS maximizes the lower bound on the probability of the observed strains simultaneously giving rise to $\dsta$, a dominant strain  $\dst$ may be estimated as a solution to the optimization problem:}
&\dst = \argmin_{y \in \cup_{\tau \leqq t} H^\tau} \sum_{x\in H^t}  \theta^{[t]}(x,y) - \abs{H^t}A \ln \mem{y}
}%
%
\subsection*{Measure of Pandemic Potential}
\def\ast{x_a^t}
\def\hst{x_h^{t+\delta}}

We measure the potential of an animal strain $\ast$ to spillover and become HH capable as a human strain $\hst$, via the proposed \erisk defined as follows:
\cgather{\label{erisk}
\rho(\ast) \triangleq -\frac{1}{\abs{H^t}} \sum_{x \in H^t} \theta^{[t]}(\ast,x)
}%
where as before $H^t$ is the set of human strains observed recently (we take this as strains collected within the past year), and $\theta^{[t]}$ is the \qdist induced by the \tnet computed from the sequences in $H^t$.

The intuition here is that a lower bound of $\rho(\ast)$ scales as average log-likelihood of the $\ast$ giving rise to a human strains in circulation at time $t$. Since the strains in $H^t$ are already HH capable, a high average likelihood of producing a similar strain has a high potential of being a HH cabale novel variant, which is a necessary condition of a pandemic strain. To establish the lower bound, we note that from  Theorem~\ref{thmbnd} (See Eq.~\eqref{fundeq} above) in the supplementary text, we have:
%
\cgather{
  \sum_{y \in H^t}\ln \abs{\frac{Pr(\ast \rightarrow y)}{\mem{y}}} \leqq -\frac{\sqrt{8}N^2}{1-\alpha} \abs{H^t}  \rho(\ast) \intertext{Denoting, $A =\frac{1-\alpha}{\sqrt{8}N^2} $,  $A\ln(\prod_{y \in H^t}\mem{y}) = C$, and $\langle \cdot \rangle$ as the geometric mean function, we have:}
\Rightarrow  \rho(\ast) \geqq A \ln \left (\prod_{y \in H^t}Pr(\ast \rightarrow y)\right )^{1/\abs{H^t}} + C \\
\Rightarrow \rho(\ast) \geqq A \ln \left \langle Pr(\ast \rightarrow \hst) \right \rangle + C
}%
Noting that $A,C$ are not functions of $\ast$, we conclude that a lower bound of the proposed risk measure $\rho(\cdot)$ scales with the average loglikelihood  of producing strains close to a circulating human strain at the current time. 

\subsection*{Proof of Probability Bounds}\label{sec:proof}

\begin{thm}[Probability bound]\label{thmbnd}
Given a sequence  $x$ of length $N$ that transitions  to a strain $y\in Q$, we have the following bounds at significance level $\alpha$.
\cgather{
\mem{y}^Q e^{ \frac{\sqrt{8}N^2}{1-\alpha}\theta(x,y)} \geqq Pr(x \rightarrow y) \geqq \mem{y}^Q e^{-\frac{\sqrt{8}N^2}{1-\alpha}\theta(x,y)}
  }%
  where $\mem{y}^Q$ is the membership probability of strain $y$ in the target population $Q$ (See Def.~\ref{defmem}), and $\theta(x,y)$ is the q-distance between $x,y$ (See Def.~\ref{defqdistance}).
\end{thm}
\begin{proof}
Using Sanov's theorem~\cite{cover} on large deviations, we conclude that the probability of spontaneous jump from strain $x\in P$ to strain $y\in Q$, with the possibility $P \neq Q$, is given by:
\cgather{\label{eq29}
  Pr(x\rightarrow y) =\prod_{i=1}^N \left ( \Phi^P_i(x_{-i}) \vert_{y_i} \right )
}
Writing the factors on the right hand side as:
\cgather{
 \Phi^P_i(x_{-i}) \vert_{y_i} =  \Phi^Q_i(y_{-i}) \vert_{y_i} \left (  \frac{\Phi^P_i(x_{-i}) \vert_{y_i}}{\Phi^Q_i(y_{-i}) \vert_{y_i}}  \right )
}%
we note that $\Phi^P_i(x_{-i})$, $\Phi^Q_i(y_{-i})$ are distributions on the same index $i$, and hence:
  \cgather{
\vert  \Phi^P_i(x_{-i})_{y_i} - \Phi^Q_i(y_{-i})_{y_i}\vert \leqq \sum_{y_i \in \Sigma_i} \vert  \Phi^P_i(x_{-i})_{y_i} - \Phi^Q_i(y_{-i})_{y_i}\vert 
}%
Using a standard refinement of Pinsker's inequality~\cite{fedotov2003refinements}, and the relationship of Jensen-Shannon divergence with  total variation, we get:
\cgather{
  \theta_i \geqq \frac{1}{8} \vert  \Phi^P_i(x_{-i})_{y_i} - \Phi^Q_i(y_{-i})_{y_i}\vert^2
\Rightarrow \left   \lvert  1  - \frac{\Phi^Q_i(y_{-i})_{y_i}}{\Phi^P_i(x_{-i})_{y_i}} \right \rvert \leqq \frac{1}{a_0}\sqrt{8 \theta_i}
}%
where $a_0$ is the smallest non-zero probability value of generating the entry at any index. We will see that this parameter is related to statistical significance of our bounds. First, we can formulate a lower bound as follows:
\cgather{\label{eqLB}
 \log \left  ( \prod_{i=1}^N   \frac{\Phi^P_i(x_{-i}) \vert_{y_i}}{\Phi^Q_i(y_{-i}) \vert_{y_i}}  \right )
  = \sum_i \log  \left  (  \frac{\Phi^P_i(x_{-i}) \vert_{y_i}}{\Phi^Q_i(y_{-i}) \vert_{y_i}}  \right )
\geqq \sum_i \left  ( 1- \frac{\Phi^Q_i(y_{-i})_{y_i}}{\Phi^P_i(x_{-i})_{y_i}} \right ) \geqq  \frac{\sqrt{8}}{a_0}\sum_i\theta_i^{1/2} = -\frac{\sqrt{8}N}{a_0}\theta
}%
Similarly,  the upper bound may be derived as:
\cgather{\label{eqUB}
\log \left  ( \prod_{i=1}^N   \frac{\Phi^P_i(x_{-i}) \vert_{y_i}}{\Phi^Q_i(y_{-i}) \vert_{y_i}}  \right )
  = \sum_i \log  \left  (  \frac{\Phi^P_i(x_{-i}) \vert_{y_i}}{\Phi^Q_i(y_{-i}) \vert_{y_i}}  \right ) \leqq \sum_i \left  ( \frac{\Phi^Q_i(y_{-i})_{y_i}}{\Phi^P_i(x_{-i})_{y_i}} - 1 \right ) \leqq \frac{\sqrt{8}N}{a_0}\theta
}%
Combining Eqs.~\ref{eqLB} and \ref{eqUB}, we conclude:
\cgather{
\mem{y}^Q e^{ \frac{\sqrt{8}N}{a_0}\theta} \geqq Pr(x \rightarrow y) \geqq \mem{y}^Q e^{-\frac{\sqrt{8}N}{a_0}\theta}
}%
Now, interpreting $a_0$ as the probability of generating an unlikely event below our desired threshold ($i.e.$ a ``failure''), we note that the probability of generating at least one such event is given by $1-(1-a_0)^N$. Hence if $\alpha$ is the pre-specified significance level, we have for $N >> 1 $:
\cgather{
 a_0 \approx (1 -\alpha)/N
}%
Hence, we conclude, that at significance level $\geqq \alpha$, we have the bounds:
\cgather{
\mem{y}^Q e^{ \frac{\sqrt{8}N^2}{1-\alpha}\theta} \geqq Pr(x \rightarrow y) \geqq \mem{y}^Q e^{-\frac{\sqrt{8}N^2}{1-\alpha}\theta}
  }%
\end{proof}
\begin{rem}
This bound can be rewritten in terms of the log-likelihood of the spontaneous jump and  constants independent of the  initial sequence $x$ as:
\cgather{
\left \lvert \log Pr(x \rightarrow y) -C_0 \right \vert \leqq C_1 \theta
}%
where the constants are given by:
\calign{
C_0 &= \log \mem{y}^Q \\
C_1 &= \frac{\sqrt{8} N^2}{1-\alpha}
}%
\end{rem}


\subsection*{In-silico Corroboration of \tnet{'s} Capability To Capture Biologically Meaningful Structure}
We compare the results of simulated mutational perturbations to sequences from our databases (for which we have already constructed \tnet{s}), and then use NCBI BLAST (\href{https://blast.ncbi.nlm.nih.gov/Blast.cgi}{https://blast.ncbi.nlm.nih.gov/Blast.cgi}) to identify  if  our perturbed sequences match with existing sequences in the databases (\SUPPLEMENTARY Fig.~S-\ref{figsoa}). We find that in contrast to random variations, which rapidly diverge the trajectories, the \tnet constraints tend to produce smaller variance in the trajectories, maintain a high degree of match as we extend our trajectories, and produces matches closer in time to the collection time of the  initial sequence, suggesting that the \tnet  does indeed capture realistic constraints.


\subsection*{Multivariate Regression to Understand Data Characteristics Necessary For \tnet Modeling}

We investigate the key factors that contribute to modeling a set of strains well within the \tnet framework. We carry out a multivariate regression with data diversity, the complexity of inferred \tnet and the edit distance of the WHO recommendation from the dominant strain as independent variables (See \SUPPLEMENTARY Table~S-\ref{tabreg} for definitions). Here we define data diversity as the number of clusters we have in the input set of sequences, such that any two sequences five or less mutations apart are in the same cluster. \tnet complexity is measured by the number of decision nodes in the component decision trees of the recursive forest.

We select several plausible structures of the regression equation, and in each case conclude that  data diversity has the most important and statistically significant contribution (\SUPPLEMENTARY Table~S-\ref{tabreg}).

\subsection*{Multivariate Regression to Identify Map from \qdist to Estimated IRAT scores}
We train separate General Linear Models (GLM) to estimate IRAT scores (emergence and impact) with average \qdist of a sequence of interest from a set of human strains, considering HA and NA sequences separately, using the CDC computed IRAT scores as the dependent variables. We also  include the geometric mean of the HA and NA based \qdist{s} as a potential explanatory variables. We use a standard Gaussian model family with identity link function to keep our model that maps \qdist{s}  to the IRAT scores as simple as possible (see \SUPPLEMENTARY Table~S-\ref{tabregGLMemergence}).


 



}